src/gsy_e/models/market/two_sided.py

```diff
@@ -79,6 +79,22 @@ def get_bids(self) -> Dict:
         """
         return deepcopy(self.bids)
 
+    def _update_requirements_prices(self, bid):
+        requirements = []
+        for requirement in bid.requirements or []:
```

Comment:
Nit: I think we could change the [init method](https://github.com/gridsingularity/gsy-framework/blob/4d6009981bae87ad63aae81f4a9bfba1d72eaf9f/gsy_framework/data_classes.py#L58) to always provide an empty list for `Bid.requirements`, so that we can avoid this check.